test/parallel/test-promises-unhandled-rejections.js

```diff
@@ -275,6 +275,20 @@ asyncTest('Attaching a promise catch in a process.nextTick is soon enough to' +
   });
 });
 
+asyncTest('Some weird edge-case', function(done) {
+  var e = new Error();
+  onUnhandledFail(done);
+
+  setImmediate(function() {
+    // reproduces on first tick and insid of setImmediate
+    Promise
+      .resolve('resolve')
```

Comment:
Probably not, the timing will be slightly different and thus the bug might not manifest